core/src/main/java/net/jonathangiles/tools/maven/whitelistgenerator/report/JSONReport.java

```diff
@@ -14,17 +16,33 @@ public String getName() {
     }
 
     @Override
-    public void report(Result result, Path reportFile) {
-        try (FileWriter writer = new FileWriter(reportFile.toFile()))  {
+    public void report(Result result, Path reportFilePath) {
+        FileWriter writer = null;
+        try {
+            File reportFile = reportFilePath.toFile();
+            if (!reportFile.exists()) {
+                System.out.println("Creating file " + reportFile.getAbsolutePath());
+                reportFile.getParentFile().mkdirs();
+                reportFile.createNewFile();
+            }
+            writer = new FileWriter(reportFile);
```

Comment:
Won't `FileWriter` create the file if it doesn't exist?

Reply:
It does only if the directory exists. It doesn't create the necessary directories.